spec/models/vm_scan_spec.rb

```diff
@@ -96,6 +99,7 @@
 
       context "in status of 'started'," do
         before(:each) do
+          MiqQueue.destroy_all
```

Comment:
I spent some time to understand the need for this line better.  The `FactoryGirl.create(:authentication...` and `@job = @vm.scan` calls in the before block of the first context are creating MiqQueue items.  The authentication creates "ems_auth_changed" and "authentication_check_types" entries and the scan creates one for "request_vm_scan".  

Since these are all created in the before block of the first context I would say the queue should be cleared at the end of that context with a comment of why it is there.  Also, you can use `MiqQueue.delete_all` and would be a little faster.

Finally, I noticed that the "when signaled with 'start'" context is using `q = MiqQueue.last` which seems like it could cause sporadic test failures if there are several items in the queue but clearing the queue resolves that as well.